public/app/features/alerting/unified/components/rules/RulesFilter.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { FormEvent, useState } from 'react';
+import { Button, Icon, Input, Label, RadioButtonGroup, useStyles } from '@grafana/ui';
+import { GrafanaTheme, SelectableValue } from '@grafana/data';
+import { css, cx } from '@emotion/css';
+import { debounce } from 'lodash';
+
+import { PromAlertingRuleState } from 'app/types/unified-alerting-dto';
+import { useQueryParams } from 'app/core/hooks/useQueryParams';
+import DataSourceSelect from '../DataSourceSelect';
+import { getFiltersFromUrlParams } from '../../utils/misc';
+
+const RulesFilter = () => {
+  const [queryParams, setQueryParams] = useQueryParams();
+  // This key is used to force a rerender on the inputs when the filters are cleared
+  const [filterKey, setFilterKey] = useState<number>(Math.floor(Math.random() * 100));
+  const dataSourceKey = `dataSource-${filterKey}`;
+  const queryStringKey = `queryString-${filterKey}`;
+
+  const { dataSource, alertState, queryString } = getFiltersFromUrlParams(queryParams);
+
+  const styles = useStyles(getStyles);
+  const stateOptions = Object.entries(PromAlertingRuleState).map(([key, value]) => ({ label: key, value }));
+
+  const handleDataSourceChange = (dataSourceValue: SelectableValue<string>) => {
+    setQueryParams({ dataSource: dataSourceValue.value });
+  };
+
+  const handleQueryStringChange = debounce((e: FormEvent<HTMLInputElement>) => {
+    const target = e.target as HTMLInputElement;
+    setQueryParams({ queryString: target.value || null });
+  }, 600);
+
+  const handleAlertStateChange = (value: string) => {
+    setQueryParams({ alertState: value });
+  };
+
+  const handleClearFiltersClick = () => {
+    setQueryParams({
+      alertState: null,
+      queryString: null,
+      dataSource: null,
+    });
+    setFilterKey(filterKey + 1);
+  };
+
+  const searchIcon = <Icon name={'search'} />;
+  return (
+    <div className={styles.container}>
+      <div className={styles.inputWidth}>
+        <Label>Select data source</Label>
+        <DataSourceSelect key={dataSourceKey} value={dataSource} onChange={handleDataSourceChange} />
+      </div>
+      <div className={cx(styles.flexRow, styles.spaceBetween)}>
+        <div className={styles.flexRow}>
+          <div className={styles.rowChild}>
+            <Label>Search by name or label</Label>
+            <Input
+              key={queryStringKey}
```

Comment:
I believe if you used `value` instead of `defaultValue`, you wouldn't need this

Reply:
Using `value` with the debounced `onChange` handler doesn't work and freezes the input after typing a few characters.

Reply:
ah yeah, because `value` does not change immediately due to debounce. One way to solve it without using `key` and remounting the component is to use `useState` to keep undebounced value. But lets's solve this later